src/shogun/machine/GeneralizedLinearMachine.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * STILL A WORK IN PROGRESS
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Tej Sukhatme
+ */
+
+// #include <rxcpp/rx-lite.hpp>
+// #include <shogun/features/DotFeatures.h>
+// #include <shogun/labels/Labels.h>
+// #include <shogun/labels/RegressionLabels.h>
+#include <shogun/machine/GeneralizedLinearMachine.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+// #include <utility>
+#include <math.h>
+#include <random>
+
+using namespace shogun;
+
+GeneralizedLinearMachine::GeneralizedLinearMachine(): Machine()
+{
+	init();
+}
+
+GeneralizedLinearMachine::GeneralizedLinearMachine(const std::shared_ptr<GeneralizedLinearMachine>& machine) : Machine()
+{
+	init();
+	require(machine, "No machine provided.");
+
+	auto w = machine->get_w();
+	auto w_clone = w.clone();
+	set_w(w_clone);
+	set_bias(machine->get_bias());
+}
+
+void GeneralizedLinearMachine::init()
+{
+	m_bias = 0;
+	features = NULL;
+	m_w = NULL;
+
+	distribution= POISSON;
+	m_alpha=0.5;
+	m_tau=NULL;
+	m_lambda=0.1;
+	m_learning_rate=2e-1;
+	m_max_iter=1000;
+	m_tolerance=1e-6;
+	m_eta=2.0;
+	m_fit_intercept=true;
+}
+
+
+GeneralizedLinearMachine::~GeneralizedLinearMachine()
+{
+
+}
+
+SGVector<float64_t> GeneralizedLinearMachine::predict(SGMatrix<float64_t> X)
+{
+	SGVector<float64_t> result = conditional_intensity(X, m_w, m_bias);
+	return result;
+}
+
+bool GeneralizedLinearMachine::fit(SGMatrix<float64_t> X, SGVector<float64_t> y)
+{
+	//Initialize parameters
+	unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
+  	std::default_random_engine generator (seed);
+	
+  	std::normal_distribution<double> distribution (0.0,1.0);
```

Comment:
mmm noup... use our internal NormalDistribution... the reason because of this is that the distr is not uniform across various stdc++ implementations.